client/dashboard/app/router/me.tsx

```diff
@@ -64,11 +64,25 @@ export const billingHistoryRoute = createRoute( {
 	)
 );
 
+export const purchaseSettingsRoute = createRoute( {
+	getParentRoute: () => meRoute,
+	loader: async ( { params: { purchaseId } } ) => {
+		await queryClient.ensureQueryData( purchaseQuery( parseInt( purchaseId ) ) );
+	},
+	path: 'billing/purchases/purchase/$purchaseId',
+} ).lazy( () =>
+	import( '../../me/billing-purchases/purchase-settings' ).then( ( d ) =>
+		createLazyRoute( 'purchases-purchase-settings' )( {
+			component: d.default,
+		} )
+	)
+);
+
 export const purchasesRoute = createRoute( {
 	getParentRoute: () => meRoute,
 	loader: async () => {
-		queryClient.ensureQueryData( userPurchasesQuery() );
-		queryClient.ensureQueryData( sitesQuery() );
+		await queryClient.ensureQueryData( userPurchasesQuery() );
```

Comment:
We should parallelize the promises. See e.g.

https://github.com/Automattic/wp-calypso/blob/eee2e436affe24a997a61af2bc0ca25f8abeacdc/client/dashboard/app/router/sites.tsx#L543-L546